server/lib/payments.js

```diff
@@ -432,18 +425,34 @@ const sendOrderConfirmedEmail = async (order, transaction) => {
 
     // hit PDF service and get PDF (unless payment method type is gift card)
     if (paymentMethod?.type !== PAYMENT_METHOD_TYPE.VIRTUALCARD) {
-      pdf = await getTransactionPdf(transaction, user);
-    }
+      const transactionPdf = await getTransactionPdf(transaction, user);
+      if (transactionPdf) {
+        const createdAtString = toIsoDateStr(transaction.createdAt ? new Date(transaction.createdAt) : new Date());
+        attachments.push({
+          filename: `transaction_${collective.slug}_${createdAtString}_${transaction.uuid}.pdf`,
+          content: transactionPdf,
+        });
+        data.transactionPdf = true;
+      }
 
-    // attach pdf
-    if (pdf) {
-      const createdAtString = toIsoDateStr(transaction.createdAt ? new Date(transaction.createdAt) : new Date());
-      attachments.push({
-        filename: `transaction_${collective.slug}_${createdAtString}_${transaction.uuid}.pdf`,
-        content: pdf,
-      });
-      data.transactionPdf = true;
+      if (transaction.hasPlatformTip()) {
+        const platformTipTransaction = await transaction.getPlatformTipTransaction();
+        if (platformTipTransaction) {
+          const platformTipPdf = await getTransactionPdf(platformTipTransaction, user);
+          if (platformTipPdf) {
+            const createdAtString = toIsoDateStr(
+              platformTipTransaction.createdAt ? new Date(platformTipTransaction.createdAt) : new Date(),
+            );
+            attachments.push({
+              filename: `transaction_opencollective_${createdAtString}_${platformTipTransaction.uuid}.pdf`,
+              content: platformTipPdf,
+            });
+            data.platformTipPdf = true;
+          }
+        }
+      }
```

Comment:
We duplicate the same code multiple times here, would be great to put that in a common helper like `attachPdfReceiptsToEmail`

Reply:
Agree and thought about it.

But it's a bit of a mess of variables and side effects and I was not able to find an elegant API for it.